tests/test_core.py

```diff
@@ -108,7 +133,29 @@ def test_read_modules_from_input_should_set_modules_attr_when_called(simple_sim)
     assert simple_sim.physics_modules[0].input_data == {"name": "ExampleModule"}
 
 
-#Grid class test methods
+def test_default_diagnostic_filename_is_generated_if_no_name_specified(simple_sim):
+    """Test read_diagnostic_from_input method in Simulation class"""
+    simple_sim.read_diagnostics_from_input()
+    input_data = simple_sim.diagnostics[0].input_data
+    assert input_data["directory"] == Path("default_output")
+    assert input_data["filename"] == str(Path("default_output")
+                                         / Path("clock0.out"))
+
+
+def test_default_diagnostic_filename_increments_for_multiple_diagnostics(simple_sim):
+    """Test read_diagnostic_from_input method in Simulation class"""
+    simple_sim.read_diagnostics_from_input()
+    assert simple_sim.diagnostics[0].input_data["directory"] == Path("default_output/")
+    assert simple_sim.diagnostics[0].input_data["filename"] == str(Path("default_output/")
```

Comment:
You don't need the trailing slashes here, since the `Path` object takes care of that for you